pkg/sources/sources.go

```diff
@@ -0,0 +1,249 @@
+	out := map[string]map[config.EventType]struct{}{}
+	for _, srcGroupCfg := range sources {
+		for _, resource := range srcGroupCfg.Kubernetes.Resources {
+			if _, ok := out[resource.Name]; !ok {
+				out[resource.Name] = make(map[config.EventType]struct{})
+			}
+			for _, e := range flattenEvents(resource.Events) {
+				out[resource.Name][e] = struct{}{}
+			}
+		}
+	}
+	return out
+}
+
+func mergeEventRoutes(resource string, sources config.IndexableMap[config.Sources]) map[config.EventType][]Route {
+	out := make(map[config.EventType][]Route)
+	for srcGroupName, srcGroupCfg := range sources {
+		for _, r := range srcGroupCfg.Kubernetes.Resources {
+			for _, e := range flattenEvents(r.Events) {
+				switch {
+				case resource == r.Name && e == config.UpdateEvent:
+					out[e] = append(out[e], Route{
+						source:     srcGroupName,
+						namespaces: r.Namespaces,
+						updateSetting: config.UpdateSetting{
+							Fields:      r.UpdateSetting.Fields,
+							IncludeDiff: r.UpdateSetting.IncludeDiff,
+						}})
+				case resource == r.Name && e != config.UpdateEvent:
+					out[e] = append(out[e], Route{source: srcGroupName, namespaces: r.Namespaces})
+				}
+			}
+		}
+	}
+	return out
+}
+
+func (r *Router) buildTable(resource string, events map[config.EventType]struct{}, routes map[config.EventType][]Route) {
+	for evt := range events {
+		if _, ok := r.table[resource]; !ok {
+
+			r.table[resource] = []Entry{{
+				event:  evt,
+				routes: routes[evt],
+			}}
+
+		} else {
+			r.table[resource] = append(r.table[resource], Entry{event: evt, routes: routes[evt]})
+		}
+	}
+}
+
+func (r *Router) RegisterInformers(targetEvents []config.EventType, handler registrationHandler) {
+	resources := r.resourcesForEvents(targetEvents)
+	for _, resource := range resources {
+		r.registrations[resource] = Registration{
+			informer:   handler(resource),
+			events:     r.resourceEvents(resource),
+			log:        r.log,
+			mapper:     r.mapper,
+			dynamicCli: r.dynamicCli,
+		}
+	}
+}
+
+func (r *Router) MapWithEventsInformer(srcEvent config.EventType, dstEvent config.EventType, handler registrationHandler) {
+	srcResources := r.resourcesForEvents([]config.EventType{srcEvent})
+	if len(srcResources) == 0 {
+		return
+	}
+
+	dstResource := "v1/events"
+	r.registrations[dstResource] = Registration{
+		informer:        handler(dstResource),
+		events:          []config.EventType{dstEvent},
+		mappedResources: srcResources,
+		mappedEvent:     srcEvent,
+		log:             r.log,
+		mapper:          r.mapper,
+		dynamicCli:      r.dynamicCli}
+}
+
+func (r *Router) HandleEvent(ctx context.Context, target config.EventType, handlerFn eventHandler) {
+	for resource, informer := range r.registrations {
+		if informer.canHandleEvent(target.String()) {
+			sourceRoutes := r.sourceRoutes(resource, target)
+			informer.handleEvent(ctx, resource, target, sourceRoutes, handlerFn)
+		}
```

Comment:
One nesting less 🙂  Same can be done with `HandleMappedEvent`.
```suggestion
		if !informer.canHandleEvent(target.String()) {
			continue
		}
		
		sourceRoutes := r.sourceRoutes(resource, target)
		informer.handleEvent(ctx, resource, target, sourceRoutes, handlerFn)
```